tbtc-ts/src/deposit.ts

```diff
@@ -1,7 +1,248 @@
-export async function createDeposit(): Promise<void> {
-  // TODO: Implementation.
+// @ts-ignore
+import bcoin from "bcoin"
+// @ts-ignore
+import { opcodes } from "bcoin/lib/script/common"
+// @ts-ignore
+import wif from "wif"
+import { BigNumber } from "ethers"
+import {
+  Client as BitcoinClient,
+  isCompressedPublicKey,
+  RawTransaction,
+  UnspentTransactionOutput,
+} from "./bitcoin"
+
+/**
+ * Contains deposit data.
+ */
+export interface DepositData {
+  /**
+   * Ethereum address prefixed with '0x' that should be used for TBTC accounting.
+   */
+  ethereumAddress: string
+
+  /**
+   * Deposit amount in sathoshis.
+   */
+  amount: BigNumber
+
+  /**
+   * Compressed (33 bytes long with 02 or 03 prefix) Bitcoin public key that
+   * is meant to be used during deposit refund after the locktime passes.
+   */
+  refundPublicKey: string
+
+  /**
+   * An 8 bytes number. Must be unique for given Ethereum address, signing group
+   * public key and refund public key.
+   */
+  blindingFactor: BigNumber
+
+  /**
+   * Unix timestamp in seconds determining the moment of deposit creation.
+   */
+  createdAt: number
 }
 
-export async function revealDeposit(): Promise<void> {
-  // TODO: Implementation.
+/**
+ * Makes a deposit by creating and broadcasting a Bitcoin P2SH
+ * deposit transaction.
+ * @param depositData - Details of the deposit.
+ * @param depositorPrivateKey - Bitcoin private key of the depositor.
+ * @param bitcoinClient - Bitcoin client used to interact with the network.
+ * @returns Empty promise.
+ */
+export async function makeDeposit(
+  depositData: DepositData,
+  depositorPrivateKey: string,
+  bitcoinClient: BitcoinClient
+): Promise<void> {
+  const depositorKeyRing = createKeyRing(depositorPrivateKey)
+  const depositorAddress = depositorKeyRing.getAddress("string")
+
+  const utxos = await bitcoinClient.findAllUnspentTransactionOutputs(
+    depositorAddress
+  )
+
+  const utxosWithRaw: (UnspentTransactionOutput & RawTransaction)[] = []
+  for (const utxo of utxos) {
+    const rawTransaction = await bitcoinClient.getRawTransaction(
```

Comment:
We will have to add error handling for communication with bitcoin, as it may always fail (I suggest we do that in another PR).

Reply:
Yep for sure but that's the matter of the implementation, not the `BitcoinClient` interface. Agree we should address it within the PR introducing Electrum client.